packages/fxa-customs-server/test/local/dataflow_tests.js

```diff
@@ -6,883 +6,678 @@ const proxyquire = require('proxyquire');
-        'missing metadata key'
-      );
-    });
+      assert.equal(log.error.callCount, 1);
+      assert.equal(log.error.args[0][0].error, 'invalid indicator: object');
 
-    test('missing entry value throws an error', () => {
-      assert.throws(
-        () =>
-          parseMessage({
-            data: Buffer.from(
-              JSON.stringify({
-                metadata: [{ key: 'bar' }],
-              })
-            ),
-          }),
-        'missing metadata value'
-      );
-    });
+      assert.equal(message.nack.callCount, 1);
 
-    test('valid message returns metadata array converted to an object', () => {
-      assert.deepEqual(
-        parseMessage({
-          data: Buffer.from(
-            JSON.stringify({
-              metadata: [
-                {
-                  key: 'foo',
-                  value: 'bar',
-                },
-                {
-                  key: 'biz',
-                  value: 'buz',
-                },
-              ],
-            })
-          ),
-        }),
-        {
-          biz: 'buz',
-          foo: 'bar',
-        }
-      );
+      assert.equal(message.ack.callCount, 0);
+      assert.equal(log.info.callCount, 0);
+      assert.equal(log.warn.callCount, 0);
+      assert.equal(fetchRecords.callCount, 0);
+      assert.equal(setRecords.callCount, 0);
     });
-  });
 
-  test('checkForUnexpectedBlock', () => {
-    let fetchSpy;
-
-    const { checkForUnexpectedBlock } = dataflow(config, log, (...args) =>
-      fetchSpy(...args)
-    );
-
-    test('missing customs_category logs an error', () => {
-      fetchSpy = sinon.spy();
+    test('empty indicator', async () => {
+      const message = {
+        ack: sandbox.spy(),
+        nack: sandbox.spy(),
+        id: 'blee',
+        data: Buffer.from(
+          JSON.stringify({
+            ...data,
+            indicator: '',
+          })
+        ),
+      };
+      await messageHandler(message);
 
-      checkForUnexpectedBlock({
-        accountid: 'testuser@testusre.com',
-        sourceaddress: '127.0.0.1',
-      });
+      assert.equal(log.error.callCount, 1);
+      assert.equal(log.error.args[0][0].error, 'invalid indicator: string');
 
-      assert.isTrue(log.error.calledOnce);
-      assert.isTrue(
-        log.error.calledOnceWith({
-          op: 'dataflow.customs_category.missing',
-        })
-      );
+      assert.equal(message.nack.callCount, 1);
 
-      assert.isFalse(fetchSpy.called);
+      assert.equal(message.ack.callCount, 0);
+      assert.equal(log.info.callCount, 0);
+      assert.equal(log.warn.callCount, 0);
+      assert.equal(fetchRecords.callCount, 0);
+      assert.equal(setRecords.callCount, 0);
     });
 
-    test('unknown category logs an error', () => {
-      fetchSpy = sinon.spy();
-
-      checkForUnexpectedBlock({
-        customs_category: 'category',
-        accountid: 'testuser@testusre.com',
-        sourceaddress: '127.0.0.1',
+    test('old message', async () => {
```

Comment:
Does 'old message' mean it was emit before the epoch value from config?

Reply:
Before `Date.now() - config.dataflow.ignoreOlderThan` specifically, yep.